packages/yoroi-extension/chrome/extension/background/handlers/yoroi/index.js

```diff
@@ -107,6 +107,7 @@ export function getHandler(typeTag: string): ?Handler {
     return async (request, send, sendResponse) => {
       try {
         const result = await handler(request.request);
+        console.debug(`BACKGROUND [${typeTag}] sending result: `, JSON.stringify(result));
         sendResponse(result);
```

Comment:
There are two different messages "channels" between thunk.js and the background:

1. API call, initialized by the front end by calling `callBackground`. Handled and replied by the background.
2. Notifications from the background to the UI. Received by `thunk.js` by calling `chrome.runtime.onMessage.addListener((serializedMessage, _sender, _sendResponse) => {` 

>  every chrome.runtime.onMessage.addListener in the thunk,js now expects to receive serializedMessage

this is 2

>  this line sendResponse(result)  

this is 1

> Should this line sendResponse(result) change to always perform serialisation?
And then remove explicit in-place serialisation where it's performed for txs?

We *can* but by now it's not necessary since everything looks OK now.

Reply:
Understood perfectly well now. Thank you!